packages/cli/src/backend-identifier/backend_identifier_resolver.ts

```diff
@@ -6,6 +6,13 @@ type BackendIdentifierParameters = {
   appId?: string;
   branch?: string;
 };
+
+export type BackendIdentityResolver = {
```

Comment:
Having `BackendIdentifierResolver` and `BackendIdentityResolver` is confusing. It's difficult to know what the different between each is. Could we have a single `BackendIdentifierResolver` type with two implementation: one that resolves a single type of BackendId and another that goes through the fallback chain?

If that proves to be too much refactoring, I think I'd rather just add the sandbox fallback directly to the existing `BackendIdentifierResolver`

Reply:
@edwardfoyle How do you feel about this? https://github.com/aws-amplify/amplify-backend/pull/669/commits/2dc8035161bada79f18f71ae6841e23d08069e3c#diff-9db562c416fe5dbc758bce7b582f0b1c080d36a408d4e2b12c457f9bbc7645b3R20